Destineo sort

kinnou02 opened the pull request:
add a custom sort for destineo: http://jira.canaltp.fr/browse/NAVITIAII-1271

kinnou02 committed:
scenario destineo: add a custom sort

kinnou02 committed:
scenario destineo: add a custom next/prev link system

the function "extremes" can now be on the scenario.
All of this need a big refactoring for a better integration of scenerios
and avoid a lot of code duplication: sometime we work with protobuf
object and other time with dict

kinnou02 committed:
scenario destineo: add tests

antoine-de commented on source/jormungandr/jormungandr/scenarios/destineo.py:
```diff
@@ -29,6 +29,164 @@
+    for section in journey.sections:
+        if section.type in (response_pb2.STREET_NETWORK, response_pb2.CROW_FLY) \
+                and section.street_network.mode in (response_pb2.Bike, response_pb2.Car, response_pb2.Bss):
+            has_non_walking = True
```
you can return early there no ?

antoine-de commented on source/jormungandr/jormungandr/scenarios/destineo.py:
```diff
@@ -29,6 +29,164 @@
+        elif is_non_pt_bike(j1):
+            return -1
+        elif is_non_pt_bike(j2):
+            return 1
```
don't you need a default return 0 ?

antoine-de commented on source/jormungandr/jormungandr/scenarios/destineo.py:
```diff
@@ -68,3 +234,30 @@ def _remove_car_if_possible(self, response):
+            list_journeys = filter(filter_journey, resp['journeys'])
+            prev_journey = min(list_journeys, key=itemgetter('arrival_date_time'))
+            next_journey = max(list_journeys, key=itemgetter('departure_date_time'))
+        except:
```
why this wide catch block ? don't you just have to check for the non emptiness of list_journeys ?

antoine-de commented on source/jormungandr/jormungandr/scenarios/destineo.py:
```diff
@@ -68,3 +234,30 @@ def _remove_car_if_possible(self, response):
+        datetime_after = None
+        prev_journey = None
+        next_journey = None
+        filter_journey = lambda journey: 'is_pure_tc' in journey['tags']
```
so for destineo if you have a query with only bss (or bike, car) you won't get prev/next ?

kinnou02 replied:
good point, I need to check the specification for that point

antoine-de commented on source/jormungandr/tests/routing_tests_destineo.py:
```diff
@@ -0,0 +1,344 @@
+# Copyright (c) 2001-2014, Canal TP and/or its affiliates. All rights reserved.
```
lot's of duplicate here, don't you think we can factorize ?

like call directly the JourneyTest methods (or find a way to copy it, but then I don't know how nose will react)

kinnou02 committed:
sceneario destineo: better testing, we don't copy all test

antoine-de commented on source/jormungandr/tests/routing_tests_destineo.py:
```diff
@@ -0,0 +1,66 @@
+import jormungandr.scenarios.destineo
+from jormungandr.instance import Instance
+
+def check_links(response):
```
useless now

kinnou02 committed:
sceneario destineo: better pagination

if there is no pure TC journeys, we create the next/prev link with all TC journeys

kinnou02 committed:
scenario destineo: add a test for destineo

antoine-de commented on source/jormungandr/tests/routing_tests_destineo.py:
```diff
@@ -0,0 +1,87 @@
+        #not to use the jormungandr database
+        query = journey_basic_query + "&first_section_mode=bss&first_section_mode=bike&first_section_mode=car" \
+                "&last_section_mode=bss&last_section_mode=bike&last_section_mode=car"
+        print query
```
old log
